lib/spack/llnl/util/lang.py

```diff
@@ -394,3 +394,15 @@ class RequiredAttributeError(ValueError):
 
     def __init__(self, message):
         super(RequiredAttributeError, self).__init__(message)
+
+
+def duplicate_stream(original):
+    """Duplicates a stream  at the os level.
+
+    :param stream original: original stream to be duplicated. Must have a
+        `fileno` callable attribute.
+
+    :return: duplicate of the original stream
+    :rtype: file like object
+    """
+    return os.fdopen(os.dup(original.fileno()))
```

Comment:
I was actually just wondering how to do this while working on an unrelated project.